### DISCUSSION TITLE
Can I tell cmake what CUDA compute capability to use?

### DISCUSSION THREAD

**Discussion by emerth** [davisking/dlib#2861] [Q&A] [answered]
Building dlib on a machine with an RTX 4070 GPU. This GPU has compute capability 8.9. Nevertheless cmake insists on using compute capability 5.0. Otherwise everything configures and compiles clean.

Can I tell cmake what compute capability to use?

Details of system:

1. Windows 11
2. Visual Studio 2019
3. RTX 4070 notebook GPU
4. Ryzen 7940HS GPU
5. CUDA 12.2
6. CUDNN 8.9.5.29_cuda12

Running the configure step shows the issue.

c:\temp\dlib\build>cmake -G "Visual Studio 16 2019" -A x64 -T host=x64 -DDLIB_USE_CUDA_COMPUTE_CAPABILITIES_STR=89 ..
-- The C compiler identification is MSVC 19.29.30151.0
-- The CXX compiler identification is MSVC 19.29.30151.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CMake version: 3.20.21032501-MSVC_2
-- Compiling dlib version: 19.24.99
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void*
-- Check size of void* - done
-- Enabling SSE2 instructions
-- Searching for BLAS and LAPACK
-- Searching for BLAS and LAPACK
-- Looking for sgesv
-- Looking for sgesv - not found
-- Looking for sgesv_
-- Looking for sgesv_ - found
-- Found CUDA: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2 (found suitable version "12.2", minimum required is "7.5")
-- Looking for cuDNN install...
-- Found cuDNN: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.2/lib/x64/cudnn.lib
-- Building a CUDA test project to see if your compiler is compatible with CUDA...
-- Building a cuDNN test project to check if you have the right version of cuDNN installed...
-- Enabling CUDA support for dlib.  **DLIB WILL USE CUDA, compute capabilities: 50**
-- Searching for FFMPEG/LIBAV
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
-- PkgConfig could not be found, FFMPEG won't be available
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/emerth/dlib/build

**Comment by arrufat** [comment, marked as answer]
You have to set the flag 
```
DLIB_USE_CUDA_COMPUTE_CAPABILITIES=89
```

On the command line:
```
-DDLIB_USE_CUDA_COMPUTE_CAPABILITIES=89
```

**Reply by arrufat** [reply]
You were setting:
```
-DDLIB_USE_CUDA_COMPUTE_CAPABILITIES_STR=89
```
Which is an internal variable not meant to be used by users.

**Reply by emerth** [reply]
Fantastic, thankyou very much!
